components/service/glean/src/test/java/mozilla/components/service/glean/ping/MetricsPingTest.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.ping
+
+import android.content.Context
+import org.junit.Test
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import java.util.concurrent.TimeUnit
+
+@RunWith(RobolectricTestRunner::class)
+class MetricsPingTest {
+    // NOTE: Using a "real" ApplicationContext here so that it will have
+    // a working SharedPreferences implementation
+    var applicationContext: Context = RuntimeEnvironment.application.applicationContext
+
+    @Test
+    fun `metrics ping has24HoursPassed() returns true if no value in shared prefs`() {
+        val metricsPing = MetricsPing(applicationContext)
+
+        // Clear the sharedPreferences on the MetricsPing so there is no last ping timestamp
+        metricsPing.metricsPreferences?.edit().clear().apply()
```

Comment:
Let's provide an `internal`, test only function in `MetricsPingScheduler` called `clearSchedulerData` that clears the prefs.

Since we want to clear the data between tests, let's add a [`@Before` rule](https://github.com/mozilla-mobile/android-components/blob/74f0ae6e7ab84f18222518aadb445c51ea0af153/components/service/glean/src/test/java/mozilla/components/service/glean/CounterMetricTypeTest.kt#L28-L39) that gets called before each test.